definitions/checks/system_registration.rb

```diff
@@ -5,15 +5,26 @@ class Checks::SystemRegistration < ForemanMaintain::Check
     tags :default
 
     confine do
-      file_exists?('/etc/rhsm/rhsm.conf') &&
-        !feature(:foreman_server) &&
-        feature(:foreman_proxy)
+      file_exists?('/etc/rhsm/rhsm.conf')
     end
   end
 
   def run
     if rhsm_hostname_eql_hostname?
-      warn! 'System is self registered'
+      if feature(:downstream)
+        notification_message = <<-MESSAGE
+  Satellite is self registered
+
+  Self registered satellite won't be supported as of Satellite version 6.3. Please follow following documentation to resolve.
```

Comment:
s/won't be/are not/

Reply:
s/Please follow following/Please follow **the** following/

Reply:
s/following documentation to resolve/following documentation to migrate/

Not required, but I think migrate reads better.